pkg/log/text_formatter.go

```diff
@@ -15,191 +15,57 @@
 package log
 
 import (
+	"bufio"
 	"bytes"
 	"fmt"
-	"runtime"
-	"sort"
 	"strings"
-	"time"
 
 	"github.com/Sirupsen/logrus"
 )
 
-const (
-	nocolor = 0
-	red     = 31
-	green   = 32
-	yellow  = 33
-	blue    = 34
-	gray    = 37
-)
-
-var (
-	baseTimestamp time.Time
-	isTerminal    bool
-)
-
-func init() {
-	baseTimestamp = time.Now()
-	isTerminal = logrus.IsTerminal()
-}
-
-func miniTS() int {
-	return int(time.Since(baseTimestamp) / time.Second)
-}
-
 type TextFormatter struct {
-
-	// Enable colors
-	Colors bool
-
-	// Enable timestamp logging. It's useful to disable timestamps when output
-	// is redirected to logging system that already adds timestamps.  When
-	// disabled, a delta is used for each log line from the start of the
-	// process.  Enabling will apply a timestamp derived from TimestampFormat.
-	Timestamp bool
-
 	// TimestampFormat is the format used to print the timestamp.  By default
 	// an RFC3339 timestamp is used.
 	TimestampFormat string
-
-	// The fields are sorted by default for a consistent output. For applications
-	// that log extremely frequently and don't use the JSON formatter this may not
-	// be desired.
-	DisableSorting bool
-
-	// Escape noncharacter ascii strings.
-	EscapeNonCharacters bool
 }
 
-// NewTextFormatter returns a text formatter with defaults appropriate for the
-// TTY or file being written to.
+// NewTextFormatter returns a text formatter
 func NewTextFormatter() *TextFormatter {
-
-	isColorTerminal := isTerminal && (runtime.GOOS != "windows")
-
-	formatter := &TextFormatter{
-		Colors:          isColorTerminal,
-		Timestamp:       false,
-		TimestampFormat: time.RFC3339,
-		DisableSorting:  false,
+	return &TextFormatter{
+		TimestampFormat: "Jan _2 2006 15:04:05.000Z07:00",
 	}
-
-	return formatter
 }
 
-func (f *TextFormatter) Format(entry *logrus.Entry) ([]byte, error) {
-	var b *bytes.Buffer
-	keys := make([]string, 0, len(entry.Data))
-	for k := range entry.Data {
-		keys = append(keys, k)
+func trimOrPadToSize(s string, size int) string {
+	if len(s) > size {
+		return s[:size]
 	}
 
-	if !f.DisableSorting {
-		sort.Strings(keys)
-	}
+	return s + strings.Repeat(" ", size-len(s))
+}
 
-	b = &bytes.Buffer{}
+func (f *TextFormatter) Format(entry *logrus.Entry) ([]byte, error) {
+	var b bytes.Buffer
 
-	// get the time string
-	ts := f.timeStamp(entry)
+	t := f.timeStamp(entry)
+	l := strings.ToUpper(trimOrPadToSize(entry.Level.String(), 4))
 
-	if f.Colors {
-		f.printColored(b, entry, keys, ts)
+	if entry.Message == "" {
+		b.WriteString(fmt.Sprintf("%s %s\n", t, l))
```

Comment:
[performance] [minor] you could directly write to b avoid another string allocation made by Sprintf